### ISSUE TITLE
Spacy dying within simple loop

### ISSUE THREAD

**Issue by andyhegedus** [explosion/spaCy#6900]
Hi,
Installed and used quickstart widget to create config file.

{'spacy_version': '3.0.1',
 'location': '/Users/andreashegedus/anaconda3/lib/python3.7/site-packages/spacy',
 'platform': 'Darwin-19.6.0-x86_64-i386-64bit',
 'python_version': '3.7.3',
 'pipelines': {'en_core_web_trf': '3.0.0'}}
Tested with simple text to an nlp object and it worked as expected.

This works as expected.
doc = nlp("Apple is looking at buying U.K. startup for $1 billion")
for token in doc:
    print(token.text, token.lemma_, token.pos_, token.tag_, token.dep_,
            token.shape_, token.is_alpha, token.is_stop)

Testing with previously successful code with Spacy 2.0 running within a Jupyterlab notebook.  The input is some text from ~2700 short documents held within object tlist.
The cell starts to run as seen by the asterisk in the JupyterLab line number field.  After ~15 sec (and this previously took several minutes to run). The kernel goes idea and the cell number with Jupyterlab is empty [].

I did some additional testing.  I modified the number of items in the input list, tlist, to 2 and it completed.  When I increased it to 3 the kernel died.

```
print(datetime.datetime.now())
stopwords = nlp.Defaults.stop_words
excludewords =['-PRON-']
excludePOS = ['SPACE','NUM','SYM','PRON']
posofword=list()
sentList=list()
pat_abstract=list()
for index,row in enumerate(tlist):
    incomingtext = row.split('.')
    for instance in incomingtext:
        incoming = nlp(instance)
        recombine=''
        word_pos=''
        for word in incoming:
            lexeme = incoming.vocab[word.text]
            if word.lemma_ not in stopwords and lexeme.is_punct==False and word.lemma_ not in excludewords and word.pos_ not in excludePOS:
                recombine += word.lemma_ +' '
                word_pos += word.pos_ +' '
        sentList.append(recombine.lower())
        posofword.append(word_pos)
        pat_abstract.append(portfolio2[index])
print(datetime.datetime.now())
```
Andy

**Comment by adrianeboyd** [comment]
Without any additional info it's hard to know what's going on for sure, but this behavior often means it's running out of memory. The transformer models like `en_core_web_trf` require a lot more memory than `en_core_web_sm`. Try using `en_core_web_sm` or `en_core_web_md` instead? For the processing here (just lemmas and POS?), the accuracy difference between `en_core_web_md` and  the huge/slow `trf` model is going to be fairly small.

You'd also see speed improvements if you rewrite your loop to use `nlp.pipe`, see: https://spacy.io/usage/processing-pipelines#processing

```python
for doc in nlp.pipe(texts):
    ...
```

**Comment by andyhegedus** [comment]
Thank you for the response.  I have tried the en_core_web_sm and it ran without issue.  The full test contains 2147 document (patents) and with the en_core_web_sm it completed it in 6.5 minutes which is what I have been seeing with Spacy 2.0.  I have also changed the cfg file to do only tagger and parser.  This did not improve the issue with en_core_web_trf; it still died quickly. This system also has 40G of ram.  I will look into pipes.
